github/login.go

```diff
@@ -83,3 +92,19 @@ func validateResponse(user *github.User, resp *github.Response, err error) error
 	}
 	return nil
 }
+
+func githubClientFromAuthURL(authURL string, httpClient *http.Client) (*github.Client, error) {
+	if strings.HasPrefix(authURL, "https://github.com/") {
+		return github.NewClient(httpClient), nil
+	} else {
+		baseURL, err := url.Parse(authURL)
+		if err != nil {
+			return nil, fmt.Errorf("could not parse AuthURL %s", authURL)
+		}
+		baseURL.Path = ""
```

Comment:
Why not just alter the baseURL.Path to be the required `/api/v3/`? Avoid the string concatenation. Pass `baseURL.String()` directly.

Also, the docs mention `Note that NewEnterpriseClient is a convenience helper only; its behavior is equivalent to using NewClient, followed by setting the BaseURL and UploadURL fields.` That may be more readable.